From 4e312a9e3bf446e88f7755ac025364ff0f2901c3 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:54 -0400
Subject: [ethernet] i40evf: don't round Tx rate down to 0

Message-id: <1409658042-22689-114-git-send-email-sassmann@redhat.com>
Patchwork-id: 90309
O-Subject: [RHEL7.1 PATCH 113/281] i40evf: don't round Tx rate down to 0
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

Because the hardware configures VF Tx rates in increments of 50 Mbps,
values smaller than that would be rounded down to 0, which was
interpreted as no limit at all. Rather than do this, we round up to 50
Mbps and notify the user.

Change-ID: I5275848233fe7514cf93e11323661c68f4c38737
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit dac9b31a18d2cd50625fd1fe3a4704dbc6d25457)

Note: changes due to missing ed616689a3d95eb6c9bdbb1ef74b0f50cbdf276a.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c b/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
index 6916ac0..36b7631 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
@@ -2251,6 +2251,11 @@ int i40e_ndo_set_vf_bw(struct net_device *netdev, int vf_id, int tx_rate)
   goto error;
  }
 
+ if (tx_rate < 50) {
+  dev_warn(&pf->pdev->dev, "Setting max Tx rate to minimum usable value of 50Mbps.\n");
+  tx_rate = 50;
+ }
+
  /* Tx rate credits are in values of 50Mbps, 0 is disabled*/
  ret = i40e_aq_config_vsi_bw_limit(&pf->hw, vsi->seid, tx_rate / 50, 0,
        NULL);
-- 
1.7.1